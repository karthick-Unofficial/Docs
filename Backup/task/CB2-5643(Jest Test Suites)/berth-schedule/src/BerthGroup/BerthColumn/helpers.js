import React from "react";

export const parseBerthData = data => {
	const parsedData = data.map(d => {
		const { beginningFootmark, endFootmark } = d;
		return {
			y0: beginningFootmark,
			y: endFootmark,
			x0: 0,
			x: 100
		};
	});
	return parsedData;
};

export const getPrimaryLabels = data => {
	const labelData = data.map(d => {
		const { endFootmark, name } = d;
		return {
			yOffset: 24,
			y: endFootmark,
			x: 8,
			label: name,
			berth: d
		};
	});
	return labelData;
};
export const getSecondaryLabels = data => {
	const labelData = data.map(d => {
		const { endFootmark, beginningFootmark } = d;
		return {
			yOffset: 44,
			y: endFootmark,
			x: 8,
			label: `${endFootmark - beginningFootmark}'`,
			style: { fontSize: 14 }
		};
	});
	return labelData;
};

export const handleTickFormat = (v, berths) => {
	const lastBerth = berths[berths.length - 1];
	const footMarks = [
		...berths.map(berth => berth.beginningFootmark),
		...berths.map(berth => berth.endFootmark)
	];
	if (footMarks.includes(v)) {
		return (
			<g>
				{v !== lastBerth.endFootmark && (
					<text
						fontSize={12}
						transform={`translate(${-40 +
							6.72 * (4 - v.toString().length)}, -10)`}
						style={{ fill: "#B5B9BE", stroke: "transparent" }}
					>
						{v}
					</text>
				)}
				<text
					fontSize={12}
					transform={`translate(${-40 + 6.72 * (4 - v.toString().length)}, 20)`}
					style={{ fill: "#B5B9BE", stroke: "transparent" }}
				>
					{v}
				</text>
			</g>
		);
	}
};

export const getIncrements = berths => {
	const lastBerth = berths[berths.length - 1];
	const increments = [];
	for (let index = 0; index < lastBerth.endFootmark; index += 100) {
		increments.push(index);
	}
	return increments;
};
