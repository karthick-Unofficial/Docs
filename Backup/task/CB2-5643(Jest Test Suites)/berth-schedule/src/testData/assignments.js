const assignments = {
    "allAssignments": {},
    "dailyAssignments": {}
};

export { assignments };