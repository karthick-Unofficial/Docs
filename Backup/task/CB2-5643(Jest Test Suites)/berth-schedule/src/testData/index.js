export { appState } from "./appState";
export { session } from "./session";
export { i18n } from "./i18n";
export { map } from "./map";
export { berths } from "./berths";
export { formPanel } from "./formPanel";
export { assignments } from "./assignments";
export { clientConfig } from "./clientConfig";