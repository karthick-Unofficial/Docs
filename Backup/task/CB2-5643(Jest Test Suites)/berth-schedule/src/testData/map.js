const map = {
    "open": false,
    "vessels": {},
    "subscriptions": []
};
export { map };