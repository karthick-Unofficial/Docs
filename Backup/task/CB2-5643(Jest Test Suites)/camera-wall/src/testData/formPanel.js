const formPanel = {
    "open": false,
    "editing": false,
    "data": {
        "agent": {
            "name": {
                "firstName": "",
                "lastName": ""
            },
            "email": "",
            "company": "",
            "phone": ""
        },
        "requestedBy": {
            "company": "",
            "name": {
                "firstName": "",
                "lastName": ""
            },
            "email": "",
            "phone": ""
        },
        "vessel": {
            "name": "",
            "mmsid": "",
            "imo": "",
            "loa": 0,
            "type": "",
            "grt": 0,
            "draft": 0
        },
        "barges": [],
        "cargo": [],
        "schedule": {
            "eta": null,
            "etd": null,
            "ata": null,
            "atd": null
        },
        "berth": {
            "id": "",
            "footmark": ""
        },
        "approved": false,
        "primaryActivity": {
            "id": ""
        }
    },
    "isGeneratingPdf": false,
    "exportingError": null
};

export { formPanel };