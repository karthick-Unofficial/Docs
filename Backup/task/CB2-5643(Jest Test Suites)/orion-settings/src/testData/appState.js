const appState = {
    "errors": {
        "newUserErrorMessage": null,
        "passwordChangeError": null,
        "shareDialogError": null,
        "createOrgError": null
    },
    "dialog": {
        "openDialog": null,
        "dialogData": null
    },
    "dock": {
        "dockData": {
            "WavCam": {}
        }
    },
    "sharingTokens": {
        "enabled": true
    },
    "viewing": {
        "selectedEntity": {
            "type": "org",
            "id": "ares_security_corporation_f2de9e6b-7f8b-4de6-b1a7-b902e083e11b"
        }
    }
};
export { appState };