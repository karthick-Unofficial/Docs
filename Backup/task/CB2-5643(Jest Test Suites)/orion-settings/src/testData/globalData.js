const globalData = {
    "orgs": {},
    "users": {
        "c1bc0cd8-9dc8-4dbb-99c1-a0300aa05700": {
            "admin": true,
            "appSettings": {
                "trackHistory": {
                    "duration": 30
                },
                "tts": {
                    "enabled": false,
                    "type": null
                }
            },
            "authProviderId": "system",
            "contact": {
                "cellPhone": null,
                "officePhone": null
            },
            "createdDate": "2023-01-13T08:00:18.692Z",
            "ecoAdmin": false,
            "email": "testorg@gm.com",
            "failedLoginAttempts": 0,
            "firstUseAck": true,
            "id": "c1bc0cd8-9dc8-4dbb-99c1-a0300aa05700",
            "lastModifiedDate": "2023-01-13T08:05:16.333Z",
            "lockedUntil": null,
            "name": "Test Org",
            "orgId": "test_organization_1",
            "orgRole": {
                "roleId": null,
                "title": "Org Admin"
            },
            "passwordChangedDate": "2023-01-13T08:02:06.726Z",
            "passwordExpires": "2023-04-13T08:02:06.726Z",
            "profileImage": "03ba4d40-9319-11ed-b28d-4bd7f7aa52bf",
            "resetExpires": null,
            "resetToken": null,
            "role": "Org User",
            "roleId": "test_organization_1_org_user",
            "setPasswordToken": null,
            "username": "testorg@gm.com"
        },
        "d7d232cb-a192-4290-83b8-a70e744a27b0": {
            "admin": false,
            "appSettings": {
                "trackHistory": {
                    "duration": 30
                },
                "tts": {
                    "enabled": false,
                    "type": null
                }
            },
            "authProviderId": "system",
            "contact": {
                "cellPhone": null,
                "officePhone": null
            },
            "createdDate": "2023-01-13T05:01:08.255Z",
            "ecoAdmin": false,
            "email": "s.karthick@rifluxyss.com",
            "failedLoginAttempts": 0,
            "firstUseAck": true,
            "id": "d7d232cb-a192-4290-83b8-a70e744a27b0",
            "lastModifiedDate": "2023-01-23T10:57:40.656Z",
            "lockedUntil": null,
            "name": "Karthick S",
            "orgId": "ares_security_corporation_f2de9e6b-7f8b-4de6-b1a7-b902e083e11b",
            "orgRole": {
                "roleId": "org-user",
                "title": "Org User"
            },
            "passwordChangedDate": "2023-01-13T05:10:07.858Z",
            "passwordExpires": "2023-04-13T05:10:07.858Z",
            "profileImage": "c192ce50-9b0c-11ed-beb9-2dd4a7b49e21",
            "resetExpires": null,
            "resetToken": null,
            "role": "Org User",
            "roleId": "ares_security_corporation_org_user",
            "setPasswordToken": null,
            "username": "s.karthick@rifluxyss.com"
        },
        "9610acee-dd03-479c-ab02-5f8fcdb0409f": {
            "admin": true,
            "appSettings": {
                "locale": null,
                "spotlightProximity": {
                    "unit": "mi",
                    "value": 0.25
                },
                "trackHistory": {
                    "duration": 30
                },
                "tts": {
                    "enabled": false,
                    "type": null
                },
                "unitsOfMeasurement": {
                    "coordinateSystem": "decimal-degrees",
                    "landUnitSystem": "imperial"
                }
            },
            "authProviderId": "system",
            "contact": {
                "address": "1501 Main Street, Suite 120 Columbia, SC 29201",
                "email": "orgadmin@aressecuritycorp.com",
                "phone": "803-339-0697"
            },
            "disabled": false,
            "ecoAdmin": false,
            "email": "orgadmin@aressecuritycorp.com",
            "id": "9610acee-dd03-479c-ab02-5f8fcdb0409f",
            "name": "Org Admin",
            "orgId": "ares_security_corporation_f2de9e6b-7f8b-4de6-b1a7-b902e083e11b",
            "role": "org-admin",
            "roleId": "ares_security_corporation_org_admin",
            "username": "orgadmin@aressecuritycorp.com"
        },
        "ed3d2e80-9737-40cd-ad06-74dc8c6e2003": {
            "admin": true,
            "appSettings": {
                "locale": "en",
                "spotlightProximity": {
                    "unit": "mi",
                    "value": 0.25
                },
                "trackHistory": {
                    "duration": 30
                },
                "tts": {
                    "enabled": false,
                    "type": null
                },
                "unitsOfMeasurement": {
                    "coordinateSystem": "decimal-degrees",
                    "landUnitSystem": "imperial"
                }
            },
            "authProviderId": "system",
            "contact": {
                "address": "1501 Main Street, Suite 120 Columbia, SC 29201",
                "cellPhone": null,
                "email": "ecoadmin@aressecuritycorp.com",
                "officePhone": null,
                "phone": "803-339-0697"
            },
            "disabled": false,
            "ecoAdmin": true,
            "email": "ecoadmin@aressecuritycorp.com",
            "failedLoginAttempts": 0,
            "firstUseAck": true,
            "id": "ed3d2e80-9737-40cd-ad06-74dc8c6e2003",
            "lastModifiedDate": "2023-01-23T06:53:35.286Z",
            "lockedUntil": null,
            "name": "Support Admin",
            "orgId": "ares_security_corporation_f2de9e6b-7f8b-4de6-b1a7-b902e083e11b",
            "profileImage": "a83a6f20-9aea-11ed-9228-07434b76e195",
            "role": "eco-admin",
            "roleId": "ares_security_corporation_org_admin",
            "username": "ecoadmin@aressecuritycorp.com"
        }
    }
};
export { globalData };