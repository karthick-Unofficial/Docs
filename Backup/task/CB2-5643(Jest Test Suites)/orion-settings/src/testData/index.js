export { appState } from "./appState";
export { session } from "./session";
export { i18n } from "./i18n";
export { globalData } from "./globalData";