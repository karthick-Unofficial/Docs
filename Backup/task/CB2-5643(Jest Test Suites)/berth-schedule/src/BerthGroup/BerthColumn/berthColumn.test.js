import React from "react";
import BerthColumn from "./BerthColumn";
import { shallow } from "enzyme";
import * as reactRedux from 'react-redux';
import { berths, i18n } from "../../testData";

jest.mock("react-redux", () => ({
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
}));

describe("BerthColumn", () => {
    beforeEach(() => {
        useSelectorMock.mockImplementation(selector => selector(mockStore));
    })
    afterEach(() => {
        useSelectorMock.mockClear();
    })

    const useSelectorMock = reactRedux.useSelector;

    const mockStore = {
        berths,
        i18n
    };

    it("renders", () => {
        const wrapper = shallow(<BerthColumn group={{ name: "" }} />);
        expect(wrapper).toMatchSnapshot();
    });
});