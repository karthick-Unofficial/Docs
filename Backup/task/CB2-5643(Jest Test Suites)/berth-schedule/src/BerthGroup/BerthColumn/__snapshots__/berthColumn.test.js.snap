// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BerthColumn renders 1`] = `ShallowWrapper {}`;
